### PR TITLE
Fix for GPD deployment

### PR DESCRIPTION
**Pull request by jacopocarlini**
The purpose of this PR is to fix configuration parameters for GPD 

### List of changes

Upd : 
  - GPD host reference
  - `SPRING_DATASOURCE` configuration

Add :  
  - `flyway_gpd.sh` script to create `apd` schema and `apd` user

**NOTE** : this PR foresees that on KV they are present :  
   - `db-apd-user-name`
   -  `db-apd-user-password`
   -  `gpd-paa-password`

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

**Review by uolter — DISMISSED** [review]

**Review by uolter — DISMISSED** [review]

File: src/main.tf [hunk 1]
```diff
@@ -8,7 +8,10 @@ terraform {
       source  = "hashicorp/azuread"
       version = "= 2.6.0"
     }
-
+    postgresql = {
+      source  = "cyrilgdn/postgresql"
```
Comment (uolter) [inline src/main.tf:12]: is it safe to use this provider? Is it kind of ufficial?

File: src/postgresql.tf [hunk 2]
```diff
@@ -9,13 +9,13 @@ module "postgresql_snet" {
 }
 
 data "azurerm_key_vault_secret" "db_administrator_login" {
-  count        = var.prostgresql_enabled ? 1 : 0
+  count        = var.postgresql_enabled ? 1 : 0
```
Comment (uolter) [inline src/postgresql.tf:12]: Do we really need this count .... why not just create the resource in every environment?

Comment (pasqualespica) [reply src/postgresql.tf:12]: unfortunately in this case yes.
In the dev environment we'll have a single server instance (used by other applications such as the mock psp too cc @infantesimone ) while in uat / prod we'll go towards a [Flexible solution](https://pagopa.atlassian.net/wiki/spaces/PAG/pages/474219118/Migrazione+DB+Oracle+su+Cloud+Azure#Tipologia-di-DB-PaaS) cc @pasqualedevita 

I changed the condition removed `postgresql_enabled`  feature flag and user `env_short` clearly
do u agree @uolter  ?

File: src/main.tf (hunk 1 above)
Comment (pasqualespica) [reply src/main.tf:12]: I removed this provider and we decided to create user/grant postegresql via flyway script

**Review by uolter — APPROVED** [review]

**Pull request merged**